docs/src/main/asciidoc/compose-dev-services.adoc

```diff
@@ -0,0 +1,632 @@
+      POSTGRES_DB: mydb
+    networks:
+      - backend
+    volumes:
+      - postgres-data:/var/lib/postgresql/data
+
+  cache:
+    image: redis:7
+    command: redis-server --save 60 1 --loglevel warning
+    ports:
+      - '6379'
+    networks:
+      - backend
+    depends_on:
+      - db
+
+  message-broker:
+    image: rabbitmq:3-management
+    ports:
+      - '5672'
+      - '15672'
+    networks:
+      - backend
+      - frontend
+    environment:
+      RABBITMQ_DEFAULT_USER: guest
+      RABBITMQ_DEFAULT_PASS: guest
+
+  api-gateway:
+    image: nginx:latest
+    ports:
+      - '8080:80'
+    networks:
+      - frontend
+    depends_on:
+      - message-broker
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf:ro
+
+networks:
+  backend:
+  frontend:
+
+volumes:
+  postgres-data:
+----
+
+Note that the `compose-devservices.yml` file can be customized as needed with multiple services with dependencies to control the startup order,
+custom networks to isolate service communication, and volumes for data persistence and configuration.
+
+== Service Discovery
+
+Services started by Compose Dev Services are automatically discovered by extensions providing dev services.
+Each extension discovers the services it needs and configures the application accordingly.
+
+Extensions rely on container image names and exposed ports to discover services.
+For example, the PostgreSQL Dev Service looks for a service with the image name *containing* `postgres` and the exposed container port `5432`.
+
+When a match is found, the extension will:
+
+1. Extract connection information from the container (connection url, credentials, database name, etc.)
+2. Configure the application to use the discovered service
+3. Skip creating its own default Dev Service container
+
+=== Supported Extensions and Discovery Criteria
+
+The following is the list of platform extensions with dev services support:
+
+[cols="1,1,1",options="header"]
+|===
+
+|Extension
+|Image Names
+|Exposed Port
+
+| AMQP
+| amqp, activemq-artemis, rabbitmq
+| 5672
+
+| Apicurio Registry
+| apicurio
+| 8080
+
+| DB2
+| db2
+| 50000
+
+| MySQL
+| mysql
+| 3306
+
+| PostgreSQL
+| postgres
+| 5432
+
+| MariaDB
+| mariadb
+| 3306
+
+| Microsoft SQL Server
+| mssql
+| 1433
+
+| Oracle Database
+| oracle
+| 1521
+
+| Kafka
+| kafka, strimzi, redpanda
+| 9092
+
+| Keycloak
+| keycloak
+| 8080
+
+| Kubernetes
+| kube-apiserver, k3s, kindest/node
+| 6443
+
+| MongoDB
+| mongo
+| 27017
+
+| MQTT
+| hivemq, eclipse-mosquitto
+| 1883
+
+| RabbitMQ
+| rabbitmq
+| 5672
+
+| Pulsar
+| pulsar
+| 6650
+
+| Redis
+| redis
+| 6379
+
+| Infinispan
+| infinispan
+| 11222
+
+| Elasticsearch
+| elasticsearch, opensearch
+| 9200
+
+| Observability
+| lgtm
+| 16686, 9411, 9090, 3000
+
+|===
+
+=== Configuring custom images for service discovery
+
+You can customize the image names used for service discovery by setting the appropriate properties in your `application.properties` file.
```

Comment:
what is this about ? isn't it using the name rather than the image name to discover services? or is this used when name does not match?

Reply:
As an example, Kafka services are discovered with image names kafka, strimzi, redpanda, according to the list above. Say you want to use a custom image `my-custom-broker:latest` in your compose file for a Kafka service. You can set the `quarkus.kafka.devservices.image-name=my-custom-broker:latest` to the custom image name, and Compose Dev Service will use it to match against services it created.